src/scene/mesh-instance.js

```diff
@@ -157,11 +157,7 @@ class MeshInstance {
 
     set mesh(mesh) {
 
-        if (mesh === this._mesh)
-            return;
-
```

Comment:
Are you sure removing this correct? I could do:

```
meshInstance.mesh = mesh;
meshInstance.mesh = mesh;
```

...and mesh will have a reference count of 2, no?

Reply:
I removed it as original code didn't have this, I added it last week.
But I'm adding it back now that you mentioned it. In general, when you call meshInstance.mesh = ..., first old mesh is decReferenced and then new one is incReferenced. It code was auto releasing mesh, then without this line, the mesh would first be dereferenced and destroyed before adding reference again.
Even though it's not needed now as meshes are not auto destroyed, it's cleaner to have it there.